### PR TITLE
rdkit: fix build for Linux

### PR DESCRIPTION
**Pull request by YoshitakaMo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I modified installation arguments for building on Linux.

### CODE REVIEW

File: Formula/rdkit.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class Rdkit < Formula
   sha256 "85cec9618e7ef6365b9b908ed674c073d898b6627521cc7fd8c2e05fea8a5def"
   license "BSD-3-Clause"
   head "https://github.com/rdkit/rdkit.git"
+  revision 1
```
Comment (carlocab) [inline Formula/rdkit.rb:8]: I don't think the revision bump is needed.

File: Formula/rdkit.rb [hunk 2]
```diff
@@ -34,7 +35,13 @@ def install
     python_executable = Formula["python@3.9"].opt_bin/"python3"
     py3ver = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     py3prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions/#{py3ver}"
+    # py3prefix on Linux is different from one on macOS
```
Comment (carlocab) [inline Formula/rdkit.rb:38]:
```suggestion
    # py3prefix on Linux is different from one on macOS
```
I think we can omit this comment, actually.

File: Formula/rdkit.rb [hunk 3]
```diff
@@ -34,7 +35,13 @@ def install
     python_executable = Formula["python@3.9"].opt_bin/"python3"
     py3ver = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     py3prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions/#{py3ver}"
+    # py3prefix on Linux is different from one on macOS 
+    on_linux do
+      py3prefix = Formula["python@3.9"].opt_prefix
+    end
     py3include = "#{py3prefix}/include/python#{py3ver}"
+    # Get homebrew numpy location
+    numpy_include = Formula["python@3.9"].opt_lib/"python#{py3ver}/site-packages/numpy/core/include"
```
Comment (carlocab) [inline Formula/rdkit.rb:44]:
```suggestion
    # Get homebrew numpy location
    numpy_include = Formula["numpy"].opt_lib/"python#{py3ver}/site-packages/numpy/core/include"
```
`numpy` is installed into its own prefix, and not Python's.

**Review by carlocab — COMMENTED** [review]
Thanks for your PR, @YoshitakaMo.

File: Formula/rdkit.rb (hunk 3 above)
Comment (YoshitakaMo) [reply Formula/rdkit.rb]: Thanks, @carlocab. I've corrected it.

File: Formula/rdkit.rb (hunk 1 above)
Comment (YoshitakaMo) [reply Formula/rdkit.rb]: `numpy` was correctly found in the previous installation on macOS.
But an additional option `-DPYTHON_NUMPY_INCLUDE_PATH=#{numpy_include}` is required for installation on Linux.

File: Formula/rdkit.rb [hunk 4]
```diff
@@ -34,7 +34,13 @@ def install
     python_executable = Formula["python@3.9"].opt_bin/"python3"
     py3ver = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     py3prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions/#{py3ver}"
+    # py3prefix on Linux is different from one on macOS
```
Comment (iMichka) [inline Formula/rdkit.rb:37]: No need for this comment

Comment (YoshitakaMo) [reply Formula/rdkit.rb]: Thanks, @iMichika. I've removed it.

File: Formula/rdkit.rb [hunk 5]
```diff
@@ -55,6 +60,7 @@ def install
       -DBoost_NO_BOOST_CMAKE=ON
       -DPYTHON_INCLUDE_DIR=#{py3include}
       -DPYTHON_EXECUTABLE=#{python_executable}
+      -DPYTHON_NUMPY_INCLUDE_PATH=#{numpy_include}
```
Comment (carlocab) [inline Formula/rdkit.rb:63]: Was the macOS build able to find `numpy` without this configuration? If it didn't, then I agree with bumping the revision. If it did, then the revision bump is not necessary.

File: Formula/rdkit.rb [hunk 6]
```diff
@@ -34,7 +34,12 @@ def install
     python_executable = Formula["python@3.9"].opt_bin/"python3"
     py3ver = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
     py3prefix = Formula["python@3.9"].opt_frameworks/"Python.framework/Versions/#{py3ver}"
+    on_linux do
+      py3prefix = Formula["python@3.9"].opt_prefix
+    end
     py3include = "#{py3prefix}/include/python#{py3ver}"
+    # Get homebrew numpy location
```
Comment (iMichka) [inline Formula/rdkit.rb:41]: Same here, the comment about numpy is not needed, you are just describing what the line does, so it does not bring any useful information. Sorry I did not notice it before.

**Review by carlocab — APPROVED** [review]
Thanks, @YoshitakaMo.

**Comment by YoshitakaMo** [comment]
I'm sorry that I don't know much about Homebrew-core, but is the bottle for linuxbrew generated automatically? Or is there something I need to do in linuxbrew-core? I would appreciate it if you could tell me.

**Comment by iMichka** [comment]
We merge from one repo to the other daily, and the bottles will be rebuilt automatically on Linux.

**Pull request merged**